.github/workflows/release.yml

```diff
@@ -37,19 +40,23 @@ jobs:
     needs: [package, test]
     runs-on: ubuntu-latest
     steps:
+    - name: git co
+      uses: actions/checkout@v1
+    - name: Get Tag Info
+      id: release-tag-meta
+      uses: ./.github/actions/release-tag-meta
+      with:
+        git-ref: ${{ github.ref }}
     - name: fetch
       uses: actions/download-artifact@v1
       with:
         name: package
         path: package
+    - runs: echo ${{ toJSON(steps) }}
     - name: release
-      uses: docker://github/ghx:master
+      uses: softprops/action-gh-release@b21b43d
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      # ghx reads GITHUB_REF and then barfs on it, no matter its command-line
-      # args. so, we override it to the SHA so it all works out.
       with:
-        entrypoint: sh
-        args: -c "export TAG=$(echo $GITHUB_REF | sed -E 's,^refs/tags/,,') NAME=$(echo $GITHUB_REF | sed -E 's,^refs/tags/release/,,') ; export GITHUB_REF=$GITHUB_SHA ; cd package && ghx release create --verbose --tag=$TAG --name=$NAME *.tar.gz"
-
-
+        name: ${{ steps.release-tag-meta.name }}
```

Comment:
...i guess you'd need a step prior to this that does something like:
```yaml
- name: Get Tag Info
  id: release-tag-meta
  run: |
    name=$(echo "$GITHUB_REF" | sed s,^refs/tags/release/,,)
    echo ::set-output "name=name::$name"
```

Reply:
I don't think you can just do this without defining an output in an action.yml... am i wrong?

Reply:
From https://help.github.com/en/articles/development-tools-for-github-actions#set-an-output-parameter-set-output

> You will receive an error if you attempt to set an output value that was not declared in the action's metadata file.

Reply:
hmm, yeah it may be possible with https://github.com/actions/github-script but that's not much simpler than rolling our own action:
https://github.community/t5/GitHub-Actions/Pass-result-of-expression-to-custom-Github-Action-argument/m-p/30458#M435